Sources/Build/misc.swift

```diff
@@ -77,9 +78,9 @@ extension ClangModule {
         
         let umbrellaHeader = Path.join(moduleHeaderDir, "\(name).h")
         if umbrellaHeader.isFile {
-            try createModuleMap(.HeaderFile)
+            try createModuleMap(inDir: wd, type: .HeaderFile)
         } else {
-            try createModuleMap(.ModuleNameDir)
+            try createModuleMap(inDir: wd, type: .ModuleNameDir)
```

Comment:
The cases here aren't exactly right. There are two different axes of choice:
 - Does the target use an "umbrella" header (defined as `\(c99name).h`), or not.
 - Does the target use a flat structure, a single subdirectory with headers, or neither (multiple subdirectories with headers).

If the target uses multiple subdirectories, we should not create a module map.

The difference between using a flat structure or a single subdirectory just changes where the module map should go. If it is a flat structure, we should produce:
```
.build/include/module.modulemap
.build/include/copied.h
```

If it is a subdirectory, we should produce:
```
.build/include/foo/module.modulemap
.build/include/foo/copied.h
```
(this is one other reason that copying may be required, so that the relative paths in the module map work out right). It probably should work out that we no longer need the -fmodule-map-file arguments when this is all done, but I am not 100% sure of that.

If the target uses an umbrella header, then the module map should be:
```
module foo {
  umbrella header "foo.h"
}
```

If the target does not use an umbrella header, then the module map should be:
module foo {
  header "a.h"
  header "b.h"
  ...
  header "xx.h"
}
```
where that lists all headers in the directory in lexicographic order.

Hopefully this is clear, unfortunately modules end up being rather complicated and we don't have a lot of great documentation about "why" for these things. Hopefully you have seen:
  http://clang.llvm.org/docs/Modules.html